src/spikeinterface/sortingcomponents/motion_interpolation.py

```diff
@@ -98,6 +101,11 @@ def interpolate_motion_on_traces(
     # assert HAVE_NUMBA
     assert times.shape[0] == traces.shape[0]
 
+    if dtype is None:
+        dtype = traces.dtype
+        if dtype.kind != "f":
+            raise ValueError(f"Can't interpolate traces of dtype {traces.dtype}.")
```

Comment:
We should allow interpolate with dtype int but with going back to int after float computation no ?

Reply:
I'm kind of confused about ints... if your dtype is int16, doesn't that usually mean you would really want to scale your recording and convert to float before interpolating? is there any use case for interpolating int data?

I'm not thrilled by the idea of a user with int16 traces who naively inputs them into the interpolation and then the data comes out looking horrible -- I think it's at least worth raising a warning in that case?

Reply:
A user with `rec.dtype == np.int16` traces could also easily `interpolate_motion(rec.astype(np.float32))` if they wanted to keep their data values the same, but I expect most people would probably be in the situation where `interpolate_motion(spre.scale(rec, gain=200.0, dtype=np.float32))` makes more sense?

Reply:
Wait... maybe those things are the same... haha. sorry. I guess I'm thinking out loud a bit here!